packages/core/src/shared/db/chatDb/chatDb.ts

```diff
@@ -371,10 +334,17 @@ export class Database {
             return
         }
 
-        const activeConversation = tabData.conversations[0]
-        let allMessages = activeConversation.messages
+        let allMessages = tabData.conversations.flatMap((conversation: Conversation) => conversation.messages)
         this.logger.info(`Found ${allMessages.length} messages in conversation`)
 
+        //  Make sure we don't exceed MaxConversationHistoryMessages
+        if (allMessages.length > MaxConversationHistoryMessages) {
+            this.logger.debug(
+                `Trimming history from ${allMessages.length} to ${MaxConversationHistoryMessages} messages`
+            )
+            allMessages = allMessages.slice(-MaxConversationHistoryMessages)
```

Comment:
this might cause invalid history? can we do findIndexToTrim

Reply:
Good callout. Fixed